Mage.Sets/src/mage/cards/r/RedemptionArc.java

```diff
@@ -0,0 +1,64 @@
+package mage.cards.r;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.ExileAttachedEffect;
+import mage.abilities.effects.common.combat.GoadAttachedEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.abilities.keyword.IndestructibleAbility;
+import mage.constants.*;
+import mage.target.common.TargetCreaturePermanent;
+import mage.abilities.effects.common.AttachEffect;
+import mage.target.TargetPermanent;
+import mage.abilities.keyword.EnchantAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class RedemptionArc extends CardImpl {
+
+    public RedemptionArc(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{W}");
+        
+        this.subtype.add(SubType.AURA);
+
+        // Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.Detriment));
+        this.addAbility(new EnchantAbility(auraTarget));
+
+        // Enchanted creature has indestructible and is goaded.
+        Effect effect = new GainAbilityAttachedEffect(IndestructibleAbility.getInstance(),
+                AttachmentType.AURA, Duration.WhileOnBattlefield);
+        effect.setText("Enchanted creature has indestructible");
+        effect.setOutcome(Outcome.Benefit);
+
+        Effect effect2 = new GoadAttachedEffect();
+        effect2.setOutcome(Outcome.Benefit);
```

Comment:
not sure about setting outcomes but only affects AI

Reply:
To be honest, i dont fully understand how the value proposition of a play is made in terms of the AI. i should have set the goaded thing to detriment by my judgement (in a general case), but setting one effect to benefit and the other to detriment on the same card seems like a strange thing to hand the computer. Is there a place this is documented?

Reply:
If ability has uses multiple effects or different outcome then you can use `Ability::addCustomOutcome` -- it will help AI with some decisions (see below about put to battlefield).

AI uses outcomes in few places like battle score calculation, targets selections, yes/no choices, etc. Basic logic:
* player need make some choice;
* AI look at effect's outcome and try make good choice: activate good effect/choice, ignore bad effect/choice
* If AI need choose targets then it will try to choose own creatures for good effects and opponent's creatures for bad effects. It it possible to cancel on bad effect without opponent targets then it will cancel it. And so on.
* Look in `CopyAITest` and `GainControlAITest` for some examples;

Also it used in some AI optimizations like put to battlefield abilities activate before combat:
![shot_240224_115617](https://github.com/magefree/mage/assets/8344157/9da190d0-4a52-490e-9df3-619c004125b1)